From 525e59246a35b9810386b2ff49a3082c303e4aa7 Mon Sep 17 00:00:00 2001
From: Fabio Erculiani <lxnay@sabayon.org>
Date: Thu, 20 Jun 2013 23:02:37 +0200
Subject: [PATCH] Add logind directories initialization and its cgroup setup

logind is part of systemd, but it will be likely used to replace
consolekit. For this reason, we expect to be able to run logind
on top of udev/openrc. logind uses dbus activation and there is
no need to manually start it.
---
 init.d/udev-mount | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/init.d/udev-mount b/init.d/udev-mount
index a868340..fc3711b 100644
--- a/init.d/udev-mount
+++ b/init.d/udev-mount
@@ -50,6 +50,17 @@ mount_dev_directory()
 	eend $rc
 }
 
+mount_logind_directories()
+{
+	export SYSTEMD_LOG_TARGET=syslog
+	checkpath -d -o root:root -m 0755 /run/systemd
+	if ! mountpoint -q /sys/fs/cgroup/systemd; then
+		checkpath -d -o root:root -m 0755 /sys/fs/cgroup/systemd
+		mount -t cgroup -o nosuid,noexec,nodev,none,name=systemd \
+			systemd /sys/fs/cgroup/systemd
+	fi
+}
+
 seed_dev()
 {
 	# Seed /dev with some things that we know we need
@@ -75,6 +86,7 @@ seed_dev()
 start()
 {
 	mount_dev_directory || return 1
+	mount_logind_directories || return 1
 
 	seed_dev
 	return 0
-- 
1.8.2.1

